newrelic/resource_newrelic_alert_compound_condition.go

```diff
@@ -0,0 +1,238 @@
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The title of the compound alert condition.",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Whether or not to enable the alert condition.",
+			},
+			"trigger_expression": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Expression that defines how component condition evaluations are combined. Valid operators are 'AND', 'OR', 'NOT'. For more complex expressions, use parentheses. Simple example: 'A AND B'. Complex example: 'A AND (B OR C) AND NOT D'.",
+			},
+			"component_conditions": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				MinItems:    2,
+				Description: "The list of NRQL conditions to be combined. Each component condition must be enabled. Must have at least 2 components.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The ID of the existing NRQL alert condition to use as a component.",
+						},
+						"alias": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The identifier that will be used in the compound alert condition's trigger_expression (e.g., 'A', 'B', 'C').",
+							ValidateFunc: validation.StringMatch(
+								regexp.MustCompile(`^[A-Za-z][A-Za-z0-9_]*$`),
+								"alias must start with a letter and contain only letters, numbers, and underscores",
+							),
+						},
+					},
+				},
+				// Custom hash to ensure uniqueness by alias
+				Set: func(v interface{}) int {
+					m := v.(map[string]interface{})
+					// Use alias for uniqueness
+					return schema.HashString(m["alias"].(string))
+				},
+			},
+			"facet_matching_behavior": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "FACETS_IGNORED",
+				Description:  "How the compound condition will take into account the component conditions' facets during evaluation. Valid values: 'FACETS_IGNORED' (default) - facets are not taken into consideration when determining when the compound alert condition activates; 'FACETS_MATCH' - the compound alert condition will activate only when shared facets have matching values.",
+				ValidateFunc: validation.StringInSlice([]string{"FACETS_MATCH", "FACETS_IGNORED"}, false),
+			},
+			"runbook_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Runbook URL to display in notifications.",
+			},
+			"threshold_duration": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "The duration, in seconds, that the trigger expression must be true before the compound alert condition will activate. Between 30-1440 seconds.",
+				ValidateFunc: validation.All(
+					validation.IntBetween(30, 1440),
+				),
```

Comment:
Not saying we have to remove this, but let's be cautious about validating in terraform vs just relying on backend errors. In case of changes in the future, it would be nice not to have to worry about updating TF